asynciterator.ts

```diff
@@ -1811,17 +1811,52 @@ class HistoryReader<T> {
   }
 }
 
+export class WrappingIterator<T> extends AsyncIterator<T> {
+  protected _source?: InternalSource<T>;
+
+  constructor(sourceOrPromise: AsyncIterator<T> | Promise<AsyncIterator<T>> | EventEmitter | Promise<EventEmitter>) {
+    super();
+    if (sourceOrPromise instanceof AsyncIterator)
+      return sourceOrPromise;
```

Comment:
As I believe you raised earlier; one of the notable things with `wrap`ing is that it will drop metadata. This modifies that behavior which may have unintended implications downstream.

Reply:
Just to make sure I understand, is the implication here that downstream objects might receive unexpected and potentially conflicting metadata?

Good catch! This would indeed be breaking behavior. Do you think it’d be worth having an option to allow this behavior, maybe set to false by default, or should we just wrap everything anyway?

Reply:
> Just to make sure I understand, is the implication here that downstream objects might receive unexpected and potentially conflicting metadata?

Yes

> Do you think it’d be worth having an option to allow this behaviour, maybe set to false by default, or should we just wrap everything anyway?

Might be a nice feature to have; but definitely false by default.

Reply:
@jeswr done!